src/teaching-element/Assessment/SingleChoice.vue

```diff
@@ -36,20 +38,30 @@ export default {
     return { userAnswer: this.submission };
   },
   computed: {
+    mappedAnswers() {
+      return this.answers.map(answer => ({
+        answer,
+        id: cuid()
+      }));
+    },
     type() {
       const options = this.options.singleChoice || defaults;
       return options.type;
     }
   },
   methods: {
-    update() {
-      this.$emit('update', { userAnswer: this.userAnswer });
-    },
-    isSelected(index) {
-      return index === this.userAnswer;
+    getAnswerClass(index) {
+      const { correct, userAnswer } = this;
+      const selected = index === userAnswer ? 'selected' : '';
+      if (!this.disabled || !this.options.enableHighlighting) return [selected];
```

Comment:
Destructure `disabled` and `options` from `this`.

Reply:
👍